django/contrib/postgres/search.py

```diff
@@ -94,11 +104,19 @@ def __init__(self, value, output_field=None, **extra):
         self.invert = extra.pop('invert', self.invert)
         super(SearchQuery, self).__init__(value, output_field=output_field)
 
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        if self.config:
+            if not hasattr(self.config, 'resolve_expression'):
+                self.config = Value(self.config)
+            self.config = self.config.resolve_expression(query, allow_joins, reuse, summarize, for_save)
```

Comment:
The private expressions needed to handle this add a lot of complexity. I've handed the issue by calling the parent's `resolve_expression()` first, which returns a copy, and then setting the `config` on the resolved expression to be either a new resolved `Value()` or `resolve_expression()` called on `self.config`, which should be a copy.